Don't use php global

imel96 opened the pull request:
Replace $_SERVER with created time argument
Add phpcbf changes

### Overview
This pull request:

- [x] Fixes a bug
- [x] Adds a feature
- [x] Breaks backwards compatibility
- [x] Needs tests

imel96 commented:
Hi guys,

This pr is still failing a few tests. Just want to know the status of this project and if I should continue works on it? Cheers

davidstrauss commented:
> This pr is still failing a few tests. Just want to know the status of this project and if I should continue works on it?

LCache is just on the back burner for me pending other projects. I think these are good changes, though.

imel96 commented:
So, we're integrating lcache with our app and we just needed to remove the php globals and on top of that we're adding a redis implementation as L2. We split the namespaces to separate L1 and L2.
Our redis patch would be based on this patch.
I see there's another PR here that's waiting to be merged, please let me know if I should rebase our patch on top of that one instead. Cheers

Commit:
Don't use php globals

Replace requested time from $_SERVER with real time
Split namespaces
Add l1, l2 and state namespaces
Minor fixes

imel96 commented:
@davidstrauss some updates, since the original PR, I've updated the patch to use time(). The reason for that is there's no other way to calculate TTL correctly without using current time. I think this it's a bug. It can be easily triggered by introducing sleep() in user code. It was discovered as part of https://github.com/imel96/simple-lcache which implements PSR SimpleCache using Lcache. I've also fixed a few other bugs and now I can confidently run SimpleLcache on production
